ni_measurementlink_generator/ni_measurementlink_generator/template.py

```diff
@@ -10,7 +10,7 @@
 def _render_template(template_name: str, **template_args) -> str:
     file_path = str(pathlib.Path(__file__).parent / "templates" / template_name)
 
-    with open(file_path, "r") as file:
+    with open(file_path, "r",encoding='UTF-8') as file:
```

Comment:
@LawsonGu running 'poetry run ni-python-styleguide fix' should fix this error and make the PR checks pass.

Reply:
I think I could not finish 'poetry run ni-python-styleguide fix' locally, this is because of this issue:
![image](https://user-images.githubusercontent.com/101655210/231642388-3cd428d7-5e08-4463-b072-565952b53c75.png)
I also noticed that CI doesn't show this issue(No module named 'pkg_resources') that is strange. After a series of check, I found that local enviornment generated from [pyproject.toml](https://github.com/ni/measurementlink-python/blob/main/pyproject.toml#L43) was required ni-python-styleguide = ">=0.4.0" but CI's version is 2.0. Which leads to local flake8 version is 5.0.4 but CI is 3.9.2. I think the CI should upgrade, but maybe local lint check is not critical and we can say our CI meet the requirements we just leave them there.

Reply:
@LawsonGu, Are you locally using Python 3.8 or .9?
The PR check uses .9, and that may workaround the conflict between sphinx and flake8 that is likely what you're hitting.

Reply:
Not it is indeed installing 2.0 among [CI](https://github.com/ni/measurementlink-python/actions/runs/4676715942/jobs/8290193818#step:8:229) . I am locally using 3.9 same as [CI: ](https://github.com/ni/measurementlink-python/actions/runs/4676715942/jobs/8290193818#step:5:6)

Reply:
Sorry for not typing right. Yeah I mean 0.2. But the poetry install from <<repo>>/pyproject.toml. And it is using >0.4.0. I think that is why I locally get the issue

Reply:
That work for running poetry install in generator's folder

Reply:
FYI, I posted a workaround for the version conflict: https://github.com/ni/measurementlink-python/pull/242

However, for this PR, reverting `poetry.lock` should be sufficient.